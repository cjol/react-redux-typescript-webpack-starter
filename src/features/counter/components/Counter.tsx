import * as React from "react";

interface Props {
  count: number;
  title: string;
  start: () => any;
}

export class Counter extends React.Component<Props, {}> {
  componentWillMount() {
    this.props.start();
  }

  render() {
    const { title, count } = this.props;
    return (
      <>
        {title && <h2>{title}</h2>}
        <div>Counter: {this.props.count}</div>
      </>
    );
  }
}
