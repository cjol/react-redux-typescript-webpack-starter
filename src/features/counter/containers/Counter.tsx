import { connect, Dispatch } from "react-redux";

import * as actions from "../actions";
import * as selectors from "../selectors";
import { counter as rootSelector } from "../../../root-selectors";
import { Counter as CounterComponent } from "../components";
import { RootState } from "../../../root-reducer";
import { bindActionCreators } from "redux";

const mapStateToProps = (root: RootState, ownProps: { title: string }) => {
  const state = rootSelector(root);

  return {
    count: selectors.getCounter(state)
  };
};

const mapDispatchToProps = {
  start: actions.initCounter
};

export const Counter = connect(mapStateToProps, mapDispatchToProps)(
  CounterComponent
);
