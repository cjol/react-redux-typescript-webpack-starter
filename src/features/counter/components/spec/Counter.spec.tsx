import * as React from "react";
import { shallow } from "enzyme";
import { Counter } from "../Counter";

describe("Counter", () => {
  const component = shallow(
    <Counter count={4} title="Sample Counter" start={() => {}} />
  );

  it("should match a snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  it("should have a title header", () => {
    const header = component.find("h2");

    expect(header.exists()).toBeTruthy();
    expect(header.contains("Sample Counter")).toBeTruthy();
  });

  it("should have a counter", () => {
    const counter = component.find("div");

    expect(counter.exists()).toBeTruthy();
    expect(counter.text()).toBe("Counter: 4");
    // expect(counter.contains("Counter: 4")).toBeTruthy();
  });
});
