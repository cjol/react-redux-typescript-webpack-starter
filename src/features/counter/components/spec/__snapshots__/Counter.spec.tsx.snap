// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    count={4}
    start={[Function]}
    title="Sample Counter"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h2>
          Sample Counter
        </h2>,
        <div>
          Counter: 
          4
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sample Counter",
        },
        "ref": null,
        "rendered": "Sample Counter",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Counter: ",
            4,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Counter: ",
          4,
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h2>
            Sample Counter
          </h2>,
          <div>
            Counter: 
            4
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sample Counter",
          },
          "ref": null,
          "rendered": "Sample Counter",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Counter: ",
              4,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Counter: ",
            4,
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
